### PR TITLE
fix(metric name): support multiple `__name__` matcher 

### LINKED ISSUE #70: Multiple `__name__` matchers not permitted
A PromQL query can contain multiple `__name__` matchers, e.g.
```
{__name__=~"node_cpu_.+", __name__=~".+total"}
```
[Here's a demo of such a query](http://demo.robustperception.io:9090/graph?g0.expr=%7B__name__%3D~%22node_cpu_.%2B%22%2C%20__name__%3D~%22.%2Btotal%22%7D&g0.tab=1&g0.stacked=0&g0.range_input=1h)

However with this lib it fails with `metric name must not be set twice: '.+total' or 'node_cpu_.+'`

Many thanks for this high quality effort btw!

### PR DESCRIPTION
**Pull request by yuanbohan**
fix #70

### CODE REVIEW

File: src/parser/promql.y [hunk 1]
```diff
@@ -389,19 +389,16 @@ unary_expr -> Result<Expr, String>:
 vector_selector -> Result<Expr, String>:
```
Comment (yuanbohan) [inline src/parser/promql.y:389]: the `__name__` matcher is absent if user does not set it explicitly, which means the metric name can be empty if neither `metric identifier` nor `__name__` matcher is set. for examples:

| expr                        | `metric identitier` | `__name__` matcher | metric |
|-----------------------------|---------------------|--------------------|--------|
| `up`                        | yes                 | no                 | up     |
| `up{foo="bar"}`             | yes                 | no                 | up     |
| `{__name__="up",foo="bar"}` | no                  | yes                | up     |
| `{foo="bar"}`               | no                  | no                 |        |

File: README.md [hunk 2]
```diff
@@ -38,9 +38,8 @@ AST: VectorSelector(VectorSelector { name: Some("http_requests_total"), matchers
 
 ## PromQL compliance
 
-This crate declares compatible with [prometheus 0372e25][prom-0372e25], which is
-prometheus release 2.40 at Nov 29, 2022. Any revision on PromQL after this
-commit is not guaranteed.
+This crate declares compatible with [prometheus v2.45.0][prom-v2.45.0], which is
+released at 2023-06-23. Any revision on PromQL after this commit is not guaranteed.
```
Comment (waynexia) [inline README.md:42]: 👍

**Review by waynexia — APPROVED** [review]
LGTM

**Review by evenyag — APPROVED** [review]
LGTM. Does greptimedb's prom planner assume that the `__name__` matcher must be present? This PR might break this assumption. @waynexia

**Comment by yuanbohan** [comment]
> LGTM. Does greptimedb's prom planner assume that the `__name__` matcher must be present? This PR might break this assumption. @waynexia

Yes，be careful to upgrade to 0.1.3 @waynexia

**Comment by waynexia** [comment]
> > LGTM. Does greptimedb's prom planner assume that the `__name__` matcher must be present? This PR might break this assumption. @waynexia
> 
> Yes，be careful to upgrade to 0.1.3 @waynexia

👍 I'll add related tests when upgrading

**Pull request merged**